protocol.go

```diff
@@ -32,7 +34,19 @@ func (p *Listener) Accept() (net.Conn, error) {
 	if err != nil {
 		return nil, err
 	}
-	return NewConn(conn), nil
+
+	proxyHeaderPolicy := USE
+	if p.Policy != nil {
+		proxyHeaderPolicy, err = p.Policy(conn.RemoteAddr())
+		if err != nil {
```

Comment:
I think this is where the issue I mentioned before is. IIRC, what we found -- using the other lib and returning an error from the checking function -- is that sometimes returning an error from Accept that wasn't a net.Error would cause a listener to simply completely shut down rather than just close the conn.

Thing is -- I can't reproduce this behavior against this code. I even tried old versions of Go, in case it was an issue fixed somewhere. So at the moment it seems okay, and maybe it was related to some specific behavior of some specific wrapping listener.

Reply:
TBH I never heard of such behavior.